### ISSUE TITLE
Install documentation consistently

### ISSUE THREAD

**Issue by danielshahaf** [zsh-users/zsh-syntax-highlighting#219]
`make install` installs the following *.md files:

```
$(PREFIX)/share/zsh-syntax-highlighting/highlighters/cursor/README.md
$(PREFIX)/share/zsh-syntax-highlighting/highlighters/README.md
$(PREFIX)/share/zsh-syntax-highlighting/highlighters/root/README.md
$(PREFIX)/share/zsh-syntax-highlighting/highlighters/main/README.md
$(PREFIX)/share/zsh-syntax-highlighting/highlighters/pattern/README.md
$(PREFIX)/share/zsh-syntax-highlighting/highlighters/line/README.md
```

It should either install the top-level `README.md` too, or install none of the other `README.md` files.

**Comment by danielshahaf** [comment]
@phy1729 points out that the *.md files should be installed to:

```make
DOCDIR?=$(DESTDIR)$(PREFIX)/share/doc/$(NAME)
```

**Comment by danielshahaf** [comment]
Draft fix: 6c54c4f (PR #232).

**Comment by danielshahaf** [comment]
Merged after review by @phy1729.
